pkg/trace/stats/quantile/ddsketch_converter.go

```diff
@@ -0,0 +1,143 @@
+		len(sketchPb.PositiveValues.BinCounts) > 0 {
+		return ddSketch{}, errors.New("ddSketch format not supported")
+	}
+	return ddSketch{
+		mapping: mapping,
+		bins: sketchPb.PositiveValues.ContiguousBinCounts,
+		offset: int(sketchPb.PositiveValues.ContiguousBinIndexOffset),
+		zeros: int(sketchPb.ZeroCount),
+	}, nil
+}
+
+func ddSketchMappingFromProto(mappingPb *pb.IndexMapping) (m mapping.IndexMapping, err error) {
+	switch mappingPb.Interpolation {
+	case pb.IndexMapping_NONE:
+		return mapping.NewLogarithmicMappingWithGamma(mappingPb.Gamma, mappingPb.IndexOffset)
+	case pb.IndexMapping_LINEAR:
+		return mapping.NewLinearlyInterpolatedMappingWithGamma(mappingPb.Gamma, mappingPb.IndexOffset)
+	case pb.IndexMapping_CUBIC:
+		return mapping.NewCubicallyInterpolatedMappingWithGamma(mappingPb.Gamma, mappingPb.IndexOffset)
+	default:
+		return nil, fmt.Errorf("interpolation not supported: %d", mappingPb.Interpolation)
+	}
+}
+
+func min(x, y int) int {
+	if x < y {
+		return x
+	}
+	return y
+}
+
+func max(x, y int) int {
+	if x > y {
+		return x
+	}
+	return y
+}
+
+// DDToGKSketches converts two dd sketches: ok and errors to 2 gk sketches: hits and errors
+// with hits = ok + errors
+func DDToGKSketches(okSketchData []byte, errSketchData []byte) (hits, errors *SliceSummary, err error) {
+	okDDSketch, err := decodeDDSketch(okSketchData)
+	if err != nil {
+		return nil, nil, err
+	}
+	// todo: remove dump
+	fmt.Println("\nok sketch")
+	spew.Dump(okDDSketch)
+	errDDSketch, err := decodeDDSketch(errSketchData)
+	if err != nil {
+		return nil, nil, err
+	}
+	// todo: remove dump
+	fmt.Println("\nerror sketch")
+	spew.Dump(errDDSketch)
+
+	minOffset := min(okDDSketch.offset, errDDSketch.offset)
+	maxIndex := max(okDDSketch.offset + len(okDDSketch.bins), errDDSketch.offset + len(errDDSketch.bins))
+	hits = &SliceSummary{Entries: make([]Entry, 0, maxIndex - minOffset)}
+	errors = &SliceSummary{Entries: make([]Entry, 0, len(errDDSketch.bins))}
+	if zeros := okDDSketch.zeros + errDDSketch.zeros; zeros > 0 {
+		hits.Entries = append(hits.Entries, Entry{V: 0, G: zeros, Delta: 0})
+		hits.N = zeros
+	}
+	if zeros := errDDSketch.zeros; zeros > 0 {
+		errors.Entries = append(errors.Entries, Entry{V: 0, G: zeros, Delta: 0})
+		errors.N = zeros
+	}
+	for index := minOffset; index < maxIndex; index++ {
+		gErr := errDDSketch.get(index)
+		gHits := okDDSketch.get(index) + gErr
+		if gHits == 0 {
+			// gHits == 0 implies gErr == 0
+			continue
+		}
+		hits.N += gHits
+		v := okDDSketch.mapping.Value(index)
+		hits.Entries = append(hits.Entries, Entry{
+			V:     v,
+			G:     gHits,
+			Delta: int(2 * EPSILON * float64(hits.N-1)),
+		})
+		if gErr == 0 {
+			continue
+		}
+		errors.N += gErr
+		errors.Entries = append(errors.Entries, Entry{
+			V:     v,
+			G:     gErr,
+			Delta: int(2 * EPSILON * float64(errors.N-1)),
+		})
+	}
+	if hits.N > 0 {
+		hits.Entries[0].Delta = 0
+		hits.Entries[len(hits.Entries)-1].Delta = 0
+	}
+	if errors.N > 0 {
```

Comment:
On the corner case of N constant values, I wonder what will happen in further merges.
If N >= 2, there should be at least a min and a max value on the GK implem